main/src/main/scala/sbt/MainLoop.scala

```diff
@@ -155,11 +155,28 @@ object MainLoop {
         state.log error errMsg
         state.fail
     }
+    val exitCode = newState.next match {
+      case State.Continue       => None
+      case State.ClearGlobalLog => None
+      case State.KeepLastLog    => None
+      case ret: State.Return =>
+        ret.result match {
+          case exit: xsbti.Exit  => Some(exit.code())
+          case _: xsbti.Continue => None
+          case _: xsbti.Reboot   => None
+          case x =>
+            val cl = if (x eq null) "" else " (class: " + x.getClass + ")"
+            newState.log debug s"Unknown main result: $x$cl"
+            None
+        }
+    }
     StandardMain.exchange publishEventMessage ExecStatusEvent(
       "Done",
       channelName,
       exec.execId,
-      newState.remainingCommands.toVector map (_.commandLine))
+      newState.remainingCommands.toVector map (_.commandLine),
+      exitCode,
```

Comment:
There's also successful event too right?
This are needs some rethinking.

Reply:
you're right.. last second change. will fix tomorrow